hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1148,31 +1148,39 @@ def example_deletion_with_block_lowering(self):
         example and every block not inside that example it tries deleting the
         example and modifying the block's value by one in either direction.
         """
-        i = 0
-        while i < len(self.shrink_target.blocks):
-            if not self.is_shrinking_block(i):
-                i += 1
-                continue
+        example_index = 0
 
-            u, v = self.blocks[i].bounds
-
-            j = 0
-            while j < len(self.shrink_target.examples):
+        while True:
+            any_examples_checked = False
```

Comment:
My understanding of this is that we don't know in advance what the largest valid example index will be.

So instead we just keep incrementing it, and then check whether it was still in-bounds for at least one block's example candidate list. If it wasn't, then we've run out of candidate block/example pairs, and we can stop the pass.